algo-builder-tester/test/v5escrow-test.ts

```diff
@@ -0,0 +1,383 @@
+      assert.equal(ASACreator.address, admin.address);
+      assert.equal(SCCreator, admin.address);
+    });
+    it("test initial global states", function () {
+      syncAccounts();
+      const sumEscrowed = admin.getGlobalState(appID, ASSET_SUM);
+      const sumIssued = admin.getGlobalState(appID, STABLE_SUM);
+      const initCRN = admin.getGlobalState(appID, "CRN");
+      assert.isDefined(sumEscrowed);
+      assert.isDefined(sumIssued);
+      assert.isDefined(initCRN);
+      assert.equal(sumEscrowed, 0n);
+      assert.equal(sumIssued, 0n);
+      assert.equal(initCRN, 5n << 32n);
+    });
+    it("assert alice opted into ASA", function () {
+      syncAccounts();
+      const last_msg: Uint8Array = alice.getLocalState(appID, "last_msg") as Uint8Array;
+      assert.equal(u8a2Str(last_msg), "OptIn OK.");
+    });
+    it("escrow 100 $ART$ to mint 100*10/5 (#$ART$*price/CR)== 200 aUSD", function () {
+      // Here both units of $ART$ and aUSD are the same, 1e-6 (by ASA.decimals).
+      const artPaid = 100n; // can be 100 (bigint is not a must)
+      // CR unit is 2^-16. 5n next line means 5>>16 * UnitCR.
+      const aUsdCollected = artPaid * 10n / 5n; // (num*price/CR)
+
+      const mintCallParams: typesW.AppCallsParam = {
+        type: typesW.TransactionType.CallApp,
+        sign: typesW.SignType.SecretKey,
+        fromAccount: alice.account,
+        appID: appID,
+        payFlags: { totalFee: fee },
+        appArgs: ['str:mint'],
+        // accounts: [admin.address], // :+1L: not working with this method, throwing error
+        // foreignAssets: [artID, usdID], // unsupported type for itxn_submit at line 211, for version 5
+      }; // TODO:ref: store a basic call params
+      const mintPayTxParams: typesW.AssetTransferParam = {
+        type: typesW.TransactionType.TransferAsset,
+        sign: typesW.SignType.SecretKey,
+        assetID: artID,
+        amount: artPaid,
+        fromAccount: alice.account,
+        toAccountAddr: admin.address,
+        payFlags: { totalFee: fee },
+      }; // TODO:ref: store a basic transfer params
+      const mintCollectTxParams: typesW.AssetTransferParam = {
+        type: typesW.TransactionType.TransferAsset,
+        sign: typesW.SignType.SecretKey,
+        assetID: usdID,
+        amount: aUsdCollected,
+        fromAccount: admin.account,
+        toAccountAddr: alice.address,
+        payFlags: { totalFee: fee },
+      }; // TODO:ref: store a basic transfer params
+      /* Check status before txn */
+      syncAccounts();
+      assert.equal(1000n, alice.assets.get(artID)!['amount']!); // 1k from dispense
+      assert.equal(999999998000n, admin.assets.get(artID)!['amount']!); // 2k dispensed
+      assert.equal(0n, alice.getLocalState(appID, ASSET_NAME)); // staked 0 $ART$ Unit
+      assert.equal(0n, alice.getLocalState(appID, STABLE_NAME)); // minted 0 aUSD Unit
+
+      const receipt = runtime.executeTx([mintCallParams, mintPayTxParams, mintCollectTxParams],);
```

Comment:
Nice setup of grouped txn! what if user only do a mintCall, without mintpay and mintcollect?

Reply:
Will be rejected by TEAL logic (length not 3). I didn't write this test. Will write one now.

Reply:
I did it in algob-tester.ts tho. but it's not clearly shown here. adding a new one.

Reply:
resolved https://github.com/Artcoin-Network/artificial-algorand-contract/commit/049055ba0f374c8f994bf038b1f137dc7ebeee3b